### ISSUE TITLE
Optional parameter with initializer is still considered potentially undefined in function body

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#10316]
```ts
function foo(x? = "bar") {
    x.toLowerCase()
}
```

Expected: No error
Actual: `Object is potentially 'undefined'.` on `x.toLowerCase()`

**Comment by ahejlsberg** [comment]
This is working as intended. To get the effect you want, simply write:

```ts
function foo(x = "bar") {
    x.toLowerCase()
}
```

This works as expected: `foo` has an optional parameter, but `x` is considered definitely assigned within the function. However, once you explicitly include a `?` on a parameter we always consider it possibly undefined, regardless of whether it has an initializer. This is analogous to how optional properties with initializers work.

**Comment by RyanCavanaugh** [comment]
It's an error to have both a `?` and a default value.
